zio-http/src/main/scala/zio/http/Scheme.scala

```diff
@@ -78,4 +89,15 @@ object Scheme       {
   case object WS extends Scheme
 
   case object WSS extends Scheme
+
+  /**
+   * @param scheme
```

Comment:
What would actually happen, if this is http(s)/ws(s)?

Reply:
Nothing
It's just a question of pattern matching.

We are allowing two representations for the same value. (`Scheme.HTTP` and `Scheme.Custom("http")`)
Since the construction method is public.
But I see this pattern is being used everywhere in zio-http so it's a more general problem

One solution would be to make it private and the only way to create Custom is with the Scheme's method `def decode(scheme: String)`.
This would make it a bit more type-safe. Instead of being on the documentation